### ISSUE TITLE
[MU4 Issue] Crashes when using arrow keys to change pitch of a note under specific circumstances

### ISSUE THREAD

**Issue by nathaniel-kite** [musescore/MuseScore#13769]
**Describe the bug**
Sometimes, adding notes to the score modifies the way the measures display on the page (this is good). If you add a note to a measure, and that measure then gets shifted to the next line, then trying to modify that note later by using the arrow keys to shift it up/down a half step will cause the program to crash.

The note remains "cursed" for as long as Musescore is open, but saving and reopening the score seems to fix the problem.

**To Reproduce**
1. Create a new score.
2. Turn on note input.
3. Fill in the first three measures with four sixteenth notes, followed by a series of eighth notes. (Image at the bottom)
4. Click on measure 4 to add a note. The score should now shift to accommodate the change, moving measure 4 to the second line.
5. Use the arrow keys to shift the new note up/down a half step.
6. Program crashes.

**Expected behavior**
Note should be shifted up/down a half step, and program should not crash.

**Platform information**
 - Windows 11

**Additional context**
It's worth noting that I've noticed many other examples of arrow keys causing crashes; this is just the only one I could actually manage to reproduce.

Here's an example where adding the note could cause problems:

![image](https://user-images.githubusercontent.com/98477315/195972420-1ef314f5-6f72-42dc-9d94-8f76d1319b66.png)

(Musescore 4 looks great, a huge improvement over Musescore 3. Keep up the great work!)

**Comment by SteveBlower** [comment]
I can't reproduce this with OS: Windows 10 Version 2009, Arch.: x86_64, MuseScore version (64-bit): 4.0.0-3254398636, revision: github-musescore-musescore-abc12345

Using the default staff space of 1.750 mm did not force a new system when I added the note at step 4.  I had to increase the space to 1.950 in Format->page settings before adding the note would force a new system (I tried both A4 and letter page sizes in case this is a US/everywhere else problem).  However, once that was done and adding the note put it on the new system moving that note using up and down arrows did not crash for me. 

Exactly which version of MS4 are you testing with?

**Comment by MarcSabatella** [comment]
What page size are you using?  For me, there is also no shift to the next measure upon entering that note, but there is if I enter several more notes.  But nothing I tried would produce a crash for me, either.

Is it possible you are using a several-months old build?  There *was* a bug where arrow key directly after a save (including autosave, so exactly once every three minutes) would crash.  But that was fixed a long time ago.  Be sure to be testing using a currently nightly build, not using an alpha build.
